### PR TITLE
Add sub-section on cert compression

### LINKED ISSUE #379: Cert compression is fine to use
§3.3. Would be good to add that TLS certificate compression is fine to use.

### PR DESCRIPTION
**Pull request by thomas-fossati**
Fix #379

Signed-off-by: Thomas Fossati <thomas.fossati@arm.com>

### CODE REVIEW

File: BCP195bis/draft-ietf-uta-rfc7525bis.md [hunk 1]
```diff
@@ -407,6 +407,22 @@ Rationale: TLS compression has been subject to security attacks, such as the CRI
...
 Implementers should note that compression at higher protocol levels can allow an active attacker to extract cleartext information from the connection. The BREACH attack is one such case. These issues can only be mitigated outside of TLS and are thus outside the scope of this document. See {{Section 2.6 of RFC7457}} for further details.
       
+### Certificate Compression
+
+Certificate chains often take up the majority of the bytes transmitted during
+the handshake.  In order to manage their size, the following methods can be
+employed:
+
+* Limiting the number of names or extensions;
+* Using keys with small public key representations, like ECDSA; or
```
Comment (stpeter) [inline BCP195bis/draft-ietf-uta-rfc7525bis.md:417]: Is this is an inclusive or? That is, can an implementation use some combination of these methods (e.g., ECDSA keys + compression)?

Comment (thomas-fossati) [reply BCP195bis/draft-ietf-uta-rfc7525bis.md]: good point. It's inclusive and I've tried to make it clear in [9ea2d9b](https://github.com/yaronf/I-D/pull/399/commits/9ea2d9b002e36dfdddb696c55bbadb900260b65c) -- which also includes some clean up

File: BCP195bis/draft-ietf-uta-rfc7525bis.md [hunk 2]
```diff
@@ -407,6 +407,22 @@ Rationale: TLS compression has been subject to security attacks, such as the CRI
...
+* Using certificate compression.
+
+For TLS 1.3 the `compress_certificate` extension is defined in {{?RFC8879}}.
+For TLS 1.2 the `cached_info` extension is defined in {{?RFC7924}}.
+
+Note that while {{?RFC8879}} compression applies to all handshakes that
+successfully negotiate the extension, {{?RFC7924}} necessitates a successful
```
Comment (yaronf) [inline BCP195bis/draft-ietf-uta-rfc7525bis.md:426]: As far as I can see, RFC 7924 is not widely implemented. For example, it is an open issue in OpenSSL. Should we even include it?

File: BCP195bis/draft-ietf-uta-rfc7525bis.md [hunk 3]
```diff
@@ -407,6 +407,28 @@ Rationale: TLS compression has been subject to security attacks, such as the CRI
...
+* For TLS 1.3, the `compress_certificate` extension is defined in {{?RFC8879}}.
+* For TLS 1.2 the `cached_info` extension is defined in {{?RFC7924}}.
+
+Note that while {{?RFC8879}} compression applies to all handshakes that
+successfully negotiate the extension, {{?RFC7924}} necessitates a successful
+full handshake before any caching can take place.
+
+See also {{Section 5 of RFC8879}} and {{Section 7 of RFC7924}} for security and
+privacy considerations associated with the use of the respective certificate
+compression technique.
```
Comment (yaronf) [inline BCP195bis/draft-ietf-uta-rfc7525bis.md:431]:
```suggestion
compression technique. To clarify: CRIME-style attacks on TLS compression do not apply to certificate compression.
```

File: BCP195bis/draft-ietf-uta-rfc7525bis.md (hunk 2 above)
Comment (thomas-fossati) [reply BCP195bis/draft-ietf-uta-rfc7525bis.md:426]: 👍 

I've done a quick survey and it looks like you are right re: support in common stacks.

**Review by yaronf — APPROVED** [review]

File: BCP195bis/draft-ietf-uta-rfc7525bis.md [hunk 4]
```diff
@@ -407,6 +407,25 @@ Rationale: TLS compression has been subject to security attacks, such as the CRI
...
 Implementers should note that compression at higher protocol levels can allow an active attacker to extract cleartext information from the connection. The BREACH attack is one such case. These issues can only be mitigated outside of TLS and are thus outside the scope of this document. See {{Section 2.6 of RFC7457}} for further details.
       
+### Certificate Compression
+
+Certificate chains often take up the majority of the bytes transmitted during
+the handshake.  In order to manage their size, some or all of the following
+methods can be employed:
+
+* Limit the number of names or extensions;
+* Use keys with small public key representations, like ECDSA; or
```
Comment (thomas-fossati) [inline BCP195bis/draft-ietf-uta-rfc7525bis.md:417]:
```suggestion
* Use keys with small public key representations, like ECDSA;
```

File: BCP195bis/draft-ietf-uta-rfc7525bis.md [hunk 5]
```diff
@@ -407,6 +407,25 @@ Rationale: TLS compression has been subject to security attacks, such as the CRI
...
+* Limit the number of names or extensions;
+* Use keys with small public key representations, like ECDSA;
+* Use certificate compression.
+
+To achieve the latter, TLS 1.3 defines the `compress_certificate` extension in
+{{?RFC8879}}.  See also {{Section 5 of RFC8879}} for security and privacy
+considerations associated with its use.  To clarify, CRIME-style attacks on TLS
+compression do not apply to certificate compression.
+
+Due to middlebox interference, RFC8879-style compression is not available in
```
[the remark this answers is not in the document]
Comment (yaronf) [reply BCP195bis/draft-ietf-uta-rfc7525bis.md:425]: Or: due to the prevalence of middlebox interference?

[the remark this answers is not in the document]
Comment (thomas-fossati) [reply BCP195bis/draft-ietf-uta-rfc7525bis.md:425]: mmm not sure.  The problem there is that security middleboxes want to snoop into the Certificate message, but RFC8879 defines a different CompressedCertificate message which, if put on the wire in clear-text (which would be the case with 1.2) would puzzle the boxes usually with catastrophic results.  So the decision was to make the extension 1.3 only.

**Review by stpeter — APPROVED** [review]

File: BCP195bis/draft-ietf-uta-rfc7525bis.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (thomas-fossati) [reply BCP195bis/draft-ietf-uta-rfc7525bis.md:425]: @stpeter @yaronf are you OK with committing the editorial suggestion above?

[the remark this answers is not in the document]
Comment (stpeter) [reply BCP195bis/draft-ietf-uta-rfc7525bis.md:425]: WFM!

[the remark this answers is not in the document]
Comment (thomas-fossati) [reply BCP195bis/draft-ietf-uta-rfc7525bis.md:425]: thanks Peter.  I'll optimistically interpret Yaron's thumbs-up in the comment above as his approval and commit that bit.

**Pull request merged**